internal/build/lifecycle_execution_test.go

```diff
@@ -1121,12 +986,45 @@ func testLifecycleExecution(t *testing.T, when spec.G, it spec.S) {
 				h.AssertSliceContains(t, configProvider.HostConfig().Binds, "/var/run/docker.sock:/var/run/docker.sock")
 			})
 
+			it("configures the phase with daemon access with TCP docker-host", func() {
+				lifecycle := newTestLifecycleExec(t, false)
+				fakePhaseFactory := fakes.NewFakePhaseFactory()
+
+				err := lifecycle.Analyze(context.Background(), "test", "test", false, "tcp://localhost:1234", false, fakeCache, fakePhaseFactory)
+				h.AssertNil(t, err)
+
+				lastCallIndex := len(fakePhaseFactory.NewCalledWithProvider) - 1
+				h.AssertNotEq(t, lastCallIndex, -1)
+
+				configProvider := fakePhaseFactory.NewCalledWithProvider[lastCallIndex]
+				h.AssertEq(t, configProvider.ContainerConfig().User, "root")
+				h.AssertSliceNotContains(t, configProvider.HostConfig().Binds, "/var/run/docker.sock:/var/run/docker.sock")
+
+				var hasDockerHostMount bool
+				for _, bind := range configProvider.HostConfig().Binds {
+					if strings.HasSuffix(bind, ":/var/run/docker.sock") {
+						hasDockerHostMount = true
+					}
+					if strings.HasSuffix(bind, `:\\.\pipe\docker_engine`) {
+						hasDockerHostMount = true
+					}
+				}
+				h.AssertTrue(t, !hasDockerHostMount)
```

Comment:
I assume we don't need both of these. Overall, the first pattern is definitely shorter – can't we just repeat it to ensure that it doesn't have the windows docker_engine bind?